src/main/java/ch/uzh/ifi/hase/soprafs22/controller/InGameWebsocketController.java

```diff
@@ -50,13 +53,29 @@ public void move(@DestinationVariable String uuid, MovePostDTO MovePostDTO, Prin
         String username = principal.getName();
         Game game = gameService.getGameByUuidOfUser(uuid, username);
 
+        if(!inGameWebsocketService.checkCanUseCard(game, move.getPlayedCard())) return;
+
+        if(move.getPlayedCard().getRank().equals(Rank.SEVEN)){
+            Ball ball = game.getBoardstate().getBallById(move.getBallId());
+            // TODO: SHould withDest be true or false??
+            List<Integer> holesTraveled = GameLogicService.getHolesTravelled(move.getDestinationTile(), ball.getPosition(), false);
+            game = inGameWebsocketService.addHolesTravelled(game, holesTraveled.size());
+        }
+
         // verify move validity, make move in game, add Player details to move for returning
         move = inGameWebsocketService.verifyMove(game, move, username);
 
         // move == null means it wasnt users turn or no cards left, simply ignore 
         if(move == null) return;
 
-        inGameWebsocketService.notifyPlayersAfterMove(game, move);
+        // Need to pass marblesset if card played was a seven
+        Set<Ball> balls = game.getBoardstate().getBalls();
+        Color userColor = game.getPlayerState(username).getColor();
+        Set<Integer> marblesSet = gameLogicService.highlightBalls(game, move.getPlayedCard().getRank(), balls, userColor, game.getColorOfTeammate(userColor));
+
+        //Not great to fetch again
+        game = gameService.getGameByUuidOfUser(uuid, username);
+        inGameWebsocketService.notifyPlayersAfterMove(game, move, marblesSet);
```

Comment:
I would suggest making the `marblesSet `an optional parameter of the `Move `class and just check inside of `notifyPlayersAfterMove` if the `marblesSet` is not empty, a seven was played and marbles have to be highlighted.